nemo/collections/asr/parts/mixins/mixins.py

```diff
@@ -390,6 +394,112 @@ def change_conv_asr_se_context_window(self, context_window: int, update_config:
+            )
+
+        (encoded, encoded_len, cache_last_channel_next, cache_last_time_next) = self.encoder.stream_step(
+            processed_signal=processed_signal,
+            processed_signal_length=processed_signal_length,
+            cache_last_channel=cache_last_channel,
+            cache_last_time=cache_last_time,
+            keep_all_outputs=keep_all_outputs,
+            drop_extra_pre_encoded=drop_extra_pre_encoded,
+        )
+
+        if isinstance(self, asr_models.EncDecCTCModel):
+            log_probs = self.decoder(encoder_output=encoded)
+            predictions_tensor = log_probs.argmax(dim=-1, keepdim=False)
+
+            # Concatenate the previous predictions with the current one to have the full predictions.
+            # We drop the extra predictions for each sample by using the lengths returned by the encoder (encoded_len)
+            # Then create a list of the predictions for the batch. The predictions can have different lengths because of the paddings.
+            greedy_predictions = []
+            if return_transcription:
+                all_hyp_or_transcribed_texts = []
+            else:
+                all_hyp_or_transcribed_texts = None
+            for preds_idx, preds in enumerate(predictions_tensor):
+                if encoded_len is None:
+                    preds_cur = predictions_tensor[preds_idx]
+                else:
+                    preds_cur = predictions_tensor[preds_idx, : encoded_len[preds_idx]]
+                if previous_pred_out is not None:
+                    greedy_predictions_concat = torch.cat((previous_pred_out[preds_idx], preds_cur), dim=-1)
+                    encoded_len[preds_idx] += len(previous_pred_out[preds_idx])
+                else:
+                    greedy_predictions_concat = preds_cur
+                greedy_predictions.append(greedy_predictions_concat)
+
+                # TODO: make decoding more efficient by avoiding the decoding process from the beginning
+                if return_transcription:
+                    decoded_out = self.decoding.ctc_decoder_predictions_tensor(
+                        decoder_outputs=greedy_predictions_concat.unsqueeze(0),
+                        decoder_lengths=encoded_len[preds_idx : preds_idx + 1],
+                        return_hypotheses=False,
+                    )
+                    all_hyp_or_transcribed_texts.append(decoded_out[0][0])
+            best_hyp = None
+        else:
+            best_hyp, all_hyp_or_transcribed_texts = self.decoding.rnnt_decoder_predictions_tensor(
+                encoder_output=encoded,
+                encoded_lengths=encoded_len,
+                return_hypotheses=True,
+                partial_hypotheses=previous_hypotheses,
+            )
+            greedy_predictions = [hyp.y_sequence for hyp in best_hyp]
+
+            if all_hyp_or_transcribed_texts is None:
+                all_hyp_or_transcribed_texts = best_hyp
+
+        return (
```

Comment:
Would it make sense to return a NamedTuple here? So many arguments can be confusing, and a named tuple normally acts like a tuple anyway.